share/config.fish

```diff
@@ -19,30 +19,32 @@ end
 
 if status --is-interactive
 	# Existance of string is a good pre-2.3.0 check. Could also check $FISH_VERSION in the future.
-
+	# This is a "launch", not an issue caused by autoloading during upgrades.
 	if not contains "string" (builtin -n)
-		set_color red;
-		echo You appear to be trying to run an old fish binary
-		echo with newer scripts installed into $__fish_datadir
-		echo This is an unsupported configuration.
+		# the string.fish message to `exec` will probably not help here, so this will that.
+		set -g __is_launched_without_string 1
+
 		set_color --bold
-		echo THINGS WILL NOT WORK.
+		echo "You appear to be trying to launch an old fish binary with newer scripts "
+		echo "installed into"; set_color --underline; echo -n "$__fish_datadir"
```

Comment:
Stylistically it's cleaner to put all commands on a new line rather than using the semicolon. And that's what fish_indent currently does.